pkg/apis/configuration/v1/types.go

```diff
@@ -77,6 +77,7 @@ type Upstream struct {
 	SlowStart                string            `json:"slow-start"`
 	Queue                    *UpstreamQueue    `json:"queue"`
 	SessionCookie            *SessionCookie    `json:"sessionCookie"`
+	GRPC                     bool              `json:"grpc"`
```

Comment:
After we reconsidered gRPC feature, we decided to go with the type field, like below:

```yaml
apiVersion: k8s.nginx.org/v1
kind: VirtualServer
metadata:
  name: cafe
spec:
  host: cafe.example.com
  tls:
    secret: cafe-secret
  upstreams:
  - name: tea
    service: tea-svc
    type: grpc # new field
    port: 50051
  routes:
  - path: /tea
    action:
      pass: tea
```

For now, the field will support two values - `http` (default) and `grpc`. Could you add the type field and remove the `grpc` field?

it will be important to add validation for that field in validateUpstreams here https://github.com/nginxinc/kubernetes-ingress/blob/1ee2bcdfa033278a9ad2ba534072cd167b8b8a1f/pkg/apis/configuration/validation/virtualserver.go#L397 . A new function like `validateUpstreamType` could do the job. 

For the documentation of the new field in the docs, I think something like below would work:
```
The type of the upstream. Supported values are ``http`` and ``grpc``. The default is ``http``. For gRPC, it is necessary to enable HTTP/2 in the `ConfigMap </nginx-ingress-controller/configuration/global-configuration/configmap-resource/#listeners)>`_ and configure TLS termination in the VirtualServer.
```